src/mantine-core/src/Hr/Hr.tsx

```diff
@@ -2,30 +2,63 @@ import React from 'react';
 import cx from 'clsx';
 import { useMantineTheme, DefaultProps, MantineNumberSize } from '@mantine/theme';
 import useStyles, { sizes } from './Hr.styles';
+import { Text, TextProps } from '../Text/Text';
 
 export const HR_SIZES = sizes;
 
 interface HrProps extends DefaultProps, React.ComponentPropsWithoutRef<'hr'> {
-  /** Hr borderStyle */
-  variant?: 'solid' | 'dashed' | 'dotted';
+  /** Hr color */
+  color?: string;
+
+  /** Applies orientation to the Hr */
+  orientation?: 'horizontal' | 'vertical';
 
   /** Hr height */
   size?: MantineNumberSize;
 
-  /** Hr color */
-  color?: string;
+  /** Applies SubHeader Text to the Hr */
+  subHeader?: string;
+
+  /** SubHeader component Props */
+  subHeaderProps?: TextProps;
+
+  /** Hr borderStyle */
+  variant?: 'solid' | 'dashed' | 'dotted';
 }
 
 export function Hr({
-  size = 'xs',
+  color = 'gray',
   className,
-  variant = 'dashed',
+  orientation = 'horizontal',
+  size = 'xs',
+  subHeader,
+  subHeaderProps,
   themeOverride,
-  color = 'gray',
+  variant = 'solid',
   ...others
 }: HrProps) {
-  const classes = useStyles({ color, variant, size, theme: useMantineTheme(themeOverride) });
-  return <hr data-mantine-hr className={cx(classes.hr, className)} {...others} />;
+  const classes = useStyles({
+    color,
+    size,
+    theme: useMantineTheme(themeOverride),
```

Comment:
Place theme as first property

Reply:
```
<Text color={color} {...subHeaderProps} className={cx(subHeaderProps?.className)}>
          {subHeader}
        </Text>
```
So finally, this would do fine?

Reply:
Just my preference for code to look cleaner
```tsx
const classes = useStyles({
  theme: useMantineTheme(themeOverride),
  color,
  size,
  // ...other
});
```